kolibri/plugins/learn/assets/src/composables/useChannels.js

```diff
@@ -1,28 +1,25 @@
 /**
  * A composable function containing logic related to channels
  */
-
+import pickBy from 'lodash/pickBy';
 import { ref, reactive } from 'kolibri.lib.vueCompositionApi';
 import { ChannelResource } from 'kolibri.resources';
 import { get, set } from '@vueuse/core';
-import plugin_data from 'plugin_data';
-
-const channelsArray = plugin_data.channels ? plugin_data.channels : [];
-const chanMap = {};
-
-for (const channel of channelsArray) {
-  chanMap[channel.id] = channel;
-}
 
 // The refs are defined in the outer scope so they can be used as a shared store
-const channels = ref(channelsArray);
-const channelsMap = reactive(chanMap);
+const channelsMap = reactive({});
+
+const localChannelsCache = ref([]);
 
 function fetchChannels(params) {
-  return ChannelResource.list(params).then(channels => {
+  params = pickBy(params || {});
+  return ChannelResource.list({ available: true, ...params }).then(channels => {
     for (const channel of channels) {
       set(channelsMap, channel.id, channel);
     }
+    if (Object.keys(params).length === 0) {
+      set(localChannelsCache, channels);
```

Comment:
The main target here is if the `baseurl` parameter is defined, in which case it is fetching channels from another device - although if there was any filtering here aside from `available: true` then this would also be problematic.